debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -137,12 +137,18 @@ public String connectionString() {
      * @throws SQLException
      */
     public SlotState getReplicationSlotInfo(String slotName, String pluginName) throws SQLException {
-        ServerInfo.ReplicationSlot slot = fetchReplicationSlotInfo(slotName, pluginName);
-        if (slot.equals(ServerInfo.ReplicationSlot.INVALID)) {
-            return null;
+        ServerInfo.ReplicationSlot slot;
+        try {
+            slot = readReplicationSlotInfo(slotName, pluginName);
+            if (slot.equals(ServerInfo.ReplicationSlot.INVALID)) {
+                return null;
```

Comment:
Why not do an if `slot == null` check? It seems you can just return the null value if that case

Reply:
This does the same without the try catch and I think is more readable. You can actually read what the valid use cases are to return valid data
```
ServerInfo.ReplicationSlot slot = fetchReplicationSlotInfo(slotName, pluginName);
if (slot != null && !slot.equals(ServerInfo.ReplicationSlot.INVALID)) {
  return slot.asSlotState();
} else {
  return null;
}

```

Reply:
@jorkzijlstra `readReplicationSlotInfo` will wait for the full initialization of the slot. The slot is already there but is not initialized yet. This is different from situation when slot is not available.

Reply:
:+1: You know the code much better than me. I was just looking at the syntax not the actual behavior up/down stream.

Reply:
A colleague of mine mentioned that when working with null objects you can also switch around the equals. In this way it doesn't crash on null objects.

Example:
```
if (!ServerInfo.ReplicationSlot.INVALID.equals(slot)) {
  return slot.asSlotState();
} else {
  return null;
}
```

In this case it probably wont help since you need the `InterruptedException`